### PR TITLE
Extract location item component

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/components/access_panels/location_item_component.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    def render_real_time_availability_request_link?
+      # we're not rendering item-level request links (because e.g. there's already alocation level request link)
+      return false unless render_item_level_request_link?
+
+      # non-circulating items don't need real time availability
+      return false unless item.circulates?
+
+      # items that definitely have at render time don't need real time availability
+      return false if item.request_link.render?
```
Comment (thatbudakguy) [inline app/components/access_panels/location_item_component.rb:48]: definitely have...?

**Review by thatbudakguy — APPROVED** [review]

File: app/components/access_panels/location_item_component.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    end
+
+    def render_real_time_availability_request_link?
+      # we're not rendering item-level request links (because e.g. there's already alocation level request link)
+      return false unless render_item_level_request_link?
+
+      # non-circulating items don't need real time availability
+      return false unless item.circulates?
+
+      # items that definitely have at render time don't need real time availability
```
Comment (cbeer) [inline app/components/access_panels/location_item_component.rb:47]:
```suggestion
      # items that definitely have an item-level request link at render time don't need real time availability
```

File: app/components/access_panels/location_item_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <%= item.loan_period %>
+      <% end %>
+      <% if render_item_level_request_link? %>
+        <span class="request-link">
+          <%= render item.request_link %>
+        </span>
+      <% end %>
+    </td>
+  <% else %>
+    <td></td>
```
Comment (cbeer) [inline app/components/access_panels/location_item_component.html.erb:33]: I hope this is the only markup difference between index/show pages; we currently show availability for bound-withs on the index page but not the show page (weird), but Sarah's ok with consistency.

File: app/components/access_panels/location_item_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<%= tag.tr class: classes, data: { barcode: item.barcode } do %>
+  <td class="callnumber">
```
Comment (cbeer) [inline app/components/access_panels/location_item_component.html.erb:2]: This PR adds css classes to the `td` elements on the index page, but they appear to be unused on the index pages. The styles seem to come from the `availability-item` class applied exclusively by the show page.

File: app/components/access_panels/location_item_component.html.erb (hunk 3 above)
Comment (thatbudakguy) [reply app/components/access_panels/location_item_component.html.erb:33]: Agree that that's weird (if anything seems like it'd be the reverse) but +1 for consistency

**Pull request merged**